File: spring_java_orm_sample/src/main/java/com/example/demo/api/controller/OpenBdBooksApiController.java
package com.example.demo.api.controller;

import com.example.demo.api.OpenBdBooksApi;
import com.example.demo.api.response.OpenBdBookResponse;
import com.example.demo.exception.OpenBdBookNotFoundException;
import com.example.demo.openbd.generated.api.BooksApi;
import com.example.demo.openbd.generated.invoker.ApiException;
import com.example.demo.openbd.generated.model.BookDto;
import lombok.RequiredArgsConstructor;
import org.modelmapper.ModelMapper;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Objects;

@RestController
@RequiredArgsConstructor
@Validated
public class OpenBdBooksApiController implements OpenBdBooksApi {
    private final BooksApi openBdBooksApi;
    private final ModelMapper modelMapper;

    @Override
    public List<OpenBdBookResponse> getBooksByIsbn(String isbn) throws ApiException {
        final var books = openBdBooksApi.getBooksByIsbn(isbn, null);
        if (Objects.isNull(books) || books.isEmpty() || books.stream().anyMatch(Objects::isNull)) {
            throw new OpenBdBookNotFoundException();
        }

        return books.stream()
            .map(this::toResponse)
            .toList();
    }

    private OpenBdBookResponse toResponse(BookDto book) {
        return modelMapper.map(book, OpenBdBookResponse.class);
    }
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/api/controller/OpenBdBooksApiController.kt
package com.example.demo.api.controller

import com.example.demo.api.OpenBdBooksApi
import com.example.demo.api.response.OpenBdBookResponse
import com.example.demo.exception.OpenBdBookNotFoundException
import com.example.demo.openbd.generated.api.BooksApi
import com.example.demo.openbd.generated.invoker.ApiException
import com.example.demo.openbd.generated.model.BookDto
import org.modelmapper.ModelMapper
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.RestController
import java.util.*

@RestController
@Validated
class OpenBdBooksApiController(private val openBdBooksApi: BooksApi, private val modelMapper: ModelMapper) :
    OpenBdBooksApi {
    @Throws(ApiException::class)
    override fun getBooksByIsbn(isbn: String): List<OpenBdBookResponse?> {
        val books = openBdBooksApi.getBooksByIsbn(isbn, null)
        if (Objects.isNull(books) || books!!.isEmpty() || books.stream()
                .anyMatch { obj: BookDto? -> Objects.isNull(obj) }
        ) {
            throw OpenBdBookNotFoundException()
        }
        return books.stream().map<OpenBdBookResponse?> { book: BookDto? -> this.toResponse(book) }.toList()
    }

    private fun toResponse(book: BookDto?): OpenBdBookResponse? {
        return modelMapper.map<OpenBdBookResponse?>(book, OpenBdBookResponse::class.java)
    }
}
